### PR TITLE
WB-1612: `IconButton`: Add PhosphorIcon support

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:
- Changes the `icon` prop to accept a `PhosphorIcon` instead of an
`IconAsset`.
- Updates the `IconButton` stories and unit tests to verify the new
  `PhosphorIcon` support.
- Updates existing WB packages to use the new `IconButton` API (Banner,
  Modal, Popover, SearchField).

**NOTE:** The plan is to replace all `IconButton` instances in webapp after this PR is landed.

Issue: WB-1612

## Test plan:

Verify that the `IconButton` stories and unit tests are passing. Also verify the
visual changes in Chromatic.

https://github.com/Khan/wonder-blocks/assets/843075/5a5c8ba1-3858-42c0-8791-085cf819ffda

### CODE REVIEW

File: __docs__/wonder-blocks-icon-button/icon-button.stories.tsx [hunk 1]
```diff
@@ -47,61 +94,41 @@ export const Default: StoryComponentType = {
     },
 };
 
-export const Basic: StoryComponentType = () => {
```
Comment (jandrade) [inline __docs__/wonder-blocks-icon-button/icon-button.stories.tsx:50]: **note:** Decided to drop this story as it is almost the same as `Default`.

File: __docs__/wonder-blocks-icon-button/icon-button.stories.tsx [hunk 2]
```diff
@@ -47,61 +94,41 @@ export const Default: StoryComponentType = {
...
+ *
+ * In this example you can see the different sizes of the icon button:
+ * - `xsmall` (16px icon with a 24px touch target).
+ * - `small` (24px icon with a 32px touch target).
+ * - `medium` (24px icon with a 40px touch target).
  */
-export const Sizes: StoryComponentType = (() => {
-    return (
-        <View style={{flexDirection: "column"}}>
+export const Sizes: StoryComponentType = {
```
Comment (jandrade) [inline __docs__/wonder-blocks-icon-button/icon-button.stories.tsx:106]: **note:** Refactored this story to present better docs formatting and take advantage of CSF v3 (Spreading stories).

File: packages/wonder-blocks-icon-button/src/__tests__/__snapshots__/custom-snapshot.test.tsx.snap
Comment (jandrade) [inline packages/wonder-blocks-icon-button/src/__tests__/__snapshots__/custom-snapshot.test.tsx.snap:1]: **note:** In this file, you can see how the markup is changed by introducing phosphor.

File: packages/wonder-blocks-popover/src/components/close-button.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import * as React from "react";
 
 import type {AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
-import {icons} from "@khanacademy/wonder-blocks-icon";
+import X from "@phosphor-icons/core/regular/x.svg";
```
[the remark this answers is not in the document]
Comment (jandrade) [reply packages/wonder-blocks-popover/src/components/close-button.tsx:4]: hmmm good point 🤔... That's right, the assets are not bundled.

My thinking is that it gets resolved by the consumer app (webapp) as `phopshor/core` is installed as a dependency, where as in `wb-modal` `phosphor/core` is installed as peer dep. For example, this gets resolved in Storybook as a static SVG asset:

e.g.
https://5e1bf4b385e3fb0020b7073c-cdjnllmslk.chromatic.com/?path=/story/modal-onepanedialog--default

The bundler (vite/rollup I guess) resolves that here: https://5e1bf4b385e3fb0020b7073c-cdjnllmslk.chromatic.com/assets/x-7397b15e.svg

I'm going to experiment more with that by consuming this pkg in webapp, and I can test without landing this PR now that Snapshot releases are coming back to the WB repo!

[the remark this answers is not in the document]
Comment (jeresig) [reply packages/wonder-blocks-popover/src/components/close-button.tsx:4]: Yeah, I think it should be pretty obvious what's happening once you get it into webapp. I think you're right that things may be ok, as it's a peer dep (as long as the bundler doesn't try to include it anyway).

[the remark this answers is not in the document]
Comment (jandrade) [reply packages/wonder-blocks-popover/src/components/close-button.tsx:4]: Experimented with that in this ZND: https://prod-znd-230929-17026-48daca0.khanacademy.org/teacher/dashboard

As a result, the icons are reused by WB and webapp:

e.g. https://cdn.kastatic.org/genwebpack/prod/en/images/4598dc228920fa0a6d94-x.svg

This is used by both the `SearchBar` component in the global header and by the WB `SearchField` component.

[the remark this answers is not in the document]
Comment (jeresig) [reply packages/wonder-blocks-popover/src/components/close-button.tsx:4]: That's great - thank you Juan!

**Pull request merged**